Fallback to HEAD and origin/master

dawidd6 opened the pull request:
Closes: #282 

In comparison to the patch attached in issue post I added this:
```ruby
... || diff_start_sha1.blank?
```
because I forgot to set `GITHUB_BASE_REF` in local Docker container and it was set in Github Actions job. So, looks like when `GITHUB_BASE_REF` is set, `diff_start_sha1` becomes blank probably because `git merge-base` returns `fatal: Not a valid object name #{diff_end_sha1}`.

With these changes I am able to successfully run `test-bot` on Github Actions.

Successful job using these changes:
https://github.com/dawidd6/homebrew-tap/runs/237362215

Unsuccessful jobs **not** using these changes:
https://github.com/dawidd6/homebrew-tap/runs/236568969
https://github.com/dawidd6/homebrew-tap/runs/237214506

Obtaining PR patch differs between these 2 unsuccessful jobs but I think it works similarly anyway.

MikeMcQuaid commented:
> because I forgot to set `GITHUB_BASE_REF` in local Docker container and it was set in Github Actions job. So, looks like when `GITHUB_BASE_REF` is set, `diff_start_sha1` becomes blank probably because `git merge-base` returns `fatal: Not a valid object name #{diff_end_sha1}`.

Can you test https://github.com/Homebrew/homebrew-test-bot/pull/284 instead? I was going to leave them as comments on this PR but I figured it might be quicker to create my own.

MikeMcQuaid commented:
Closing in favour of https://github.com/Homebrew/homebrew-test-bot/pull/284 but happy to keep talking about this if it doesn't address the issue(s).

dawidd6 commented:
It slightly improved with your patch, thanks, but it's still not this.

Current situation:
```
Testing tap dawidd6/tap with:
  origin/master   38b267e (workflow: test upstream test-bot)
  HEAD            b571f31 (neofetch: test build with upstream test-bot)
  diff_start_sha1 b571f31
  diff_end_sha1   34a0815c9ed21a0aebe8eb2b7617a83366aa6098

Formula changes to be tested:
  added formulae    (empty)
  modified formulae (empty)
  deleted formulae  (empty)
```

Previous proper:
```
Testing tap dawidd6/tap with:
  origin/master   e32c27e (workflow: handle multiple packages)
  HEAD            07f46b5 (neofetch: test)
  diff_start_sha1 e32c27e
  diff_end_sha1   07f46b5

Formula changes to be tested:
  added formulae    (empty)
  modified formulae dawidd6/tap/neofetch
  deleted formulae  (empty)
```

MikeMcQuaid commented:
You need to pass through the following variables: https://github.com/Homebrew/brew/blob/2c7091f06d801502bc0e55e4b9b876ff3007619d/Dockerfile.yml#L19-L26

Have you passed through all of those correctly? If not, I'll probably improve the error messaging again.

dawidd6 commented:
Just made a check:
https://github.com/dawidd6/homebrew-tap/runs/237767418

```
Run echo "GITHUB_ACTIONS    $GITHUB_ACTIONS"
  echo "GITHUB_ACTIONS    $GITHUB_ACTIONS"
  echo "GITHUB_BASE_REF   $GITHUB_BASE_REF"
  echo "GITHUB_EVENT_NAME $GITHUB_EVENT_NAME"
  echo "GITHUB_REF        $GITHUB_REF"
  echo "GITHUB_REPOSITORY $GITHUB_REPOSITORY"
  echo "GITHUB_SHA        $GITHUB_SHA"
  echo "HEAD_GITHUB_REF   $HEAD_GITHUB_REF"
  shell: /bin/bash -e {0}
GITHUB_ACTIONS    true
GITHUB_BASE_REF   master
GITHUB_EVENT_NAME pull_request
GITHUB_REF        refs/pull/37/merge
GITHUB_REPOSITORY dawidd6/homebrew-tap
GITHUB_SHA        6aa743b1aa236bf4079b5dbc2aa42f0018adbd3c
HEAD_GITHUB_REF   
```

It seems like I don't have `HEAD_GITHUB_REF` set, but judging by code of `test-bot` it is not used anywhere.
